app/src/main/java/com/hedvig/app/util/featureflags/RemoteFeatureFlagProvider.kt

```diff
@@ -0,0 +1,46 @@
+package com.hedvig.app.util.featureflags
+
+import com.hedvig.app.feature.settings.Market
+import com.hedvig.app.feature.settings.MarketManager
+import com.hedvig.app.service.RemoteConfig
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+class RemoteFeatureFlagProvider(
+    private val marketManager: MarketManager,
+    private val remoteConfig: RemoteConfig,
+) : FeatureFlagProvider {
+
+    override val priority = REMOTE_PRIORITY
+
+    private var seCampaignVisible: Boolean = false
+    private var noCampaignVisible: Boolean = false
+    private var dkCampaignVisible: Boolean = false
+
+    init {
+        CoroutineScope(Dispatchers.IO).launch {
+            seCampaignVisible = remoteConfig.fetch().seCampaignVisible
+            noCampaignVisible = remoteConfig.fetch().noCampaignVisible
+            dkCampaignVisible = remoteConfig.fetch().dkCampaignVisible
```

Comment:
Doesn't fetch() return an object that contains all the variables by itself? Here we're doing 3 fetches but all three will make a network request as I understand right? If it can be 1 request it'd be better to be like that. If for some reason the have to be separate requests it'd be a good idea to wrap them all in `async {}` and put them in a list and do `.awaitAll()` on them so that they run in parallel at least instead of sequentially.

Reply:
Yeah will definitely want to only do one call here 😅 My mistake!